### PR TITLE
PTL-3452 - Adding server and client versions v2 and v3 to support if data in signature

### PR DESCRIPTION
**Pull request by mgoodnow**
- Updating README documentation
- Updating HURL cli to support data signing or not

### CODE REVIEW

File: server2/src/main/java/com/bazaarvoice/auth/hmac/server/PrincipalFactory.java [hunk 1]
```diff
@@ -39,19 +35,23 @@
...
 
     /**
      * @param authenticator the application's credential authenticator (required)
      * @param requestProvider object that provides access to the active request (required)
      */
     @Inject
     public PrincipalFactory(final Authenticator<P> authenticator,
-            final Provider<ContainerRequest> requestProvider) {
+                            final Provider<ContainerRequest> requestProvider,
+                            final RequestConfiguration requestConfiguration) {
```
Comment (jennifergarcia) [inline server2/src/main/java/com/bazaarvoice/auth/hmac/server/PrincipalFactory.java]: This new parameter needs to be added to the javadoc

File: cli/src/main/java/com/bazaarvoice/auth/hmac/cli/HurlCli.java [hunk 2]
```diff
@@ -31,7 +31,6 @@ public static void main(String[] args) {
         parser.addArgument("-X", "--request").help("GET (default), POST, PUT, or DELETE").type(String.class).choices("GET", "POST", "PUT", "DELETE").setDefault("GET");
         parser.addArgument("--apiKey").required(true);
         parser.addArgument("--secretKey").required(true);
-        parser.addArgument("--signData").required(false);
```
Comment (jennifergarcia) [inline cli/src/main/java/com/bazaarvoice/auth/hmac/cli/HurlCli.java]: I think that we need to add --version on here right?

File: common/src/main/java/com/bazaarvoice/auth/hmac/common/Version.java [hunk 3]
```diff
@@ -1,21 +1,43 @@
...
 /**
- * Represents a version of the authentication contract between the client and server
+ * Represents a version of the authentication contract between the client and server.
  */
 public enum Version {
-    V1("1");
+
+    /**
+     * Version 1
+     * <p> - Has mix implementations for including content body in signature
```
Comment (jennifergarcia) [inline common/src/main/java/com/bazaarvoice/auth/hmac/common/Version.java]: typo grammar,  mix -> mixed

File: cli/src/main/java/com/bazaarvoice/auth/hmac/cli/HurlCli.java (hunk 2 above)
Comment (mgoodnow) [reply cli/src/main/java/com/bazaarvoice/auth/hmac/cli/HurlCli.java]: Done

**Comment by mgoodnow** [comment]
For what versions server accepts, updated to either use defaults or require specifying all versions supporting.

**Pull request merged**